### PR TITLE
Use JShell with Bach.java to build and run the project on any platform

### PR DESCRIPTION
**Pull request by sormuras**
This PR introduces a Java Shell Builder named `Bach.java`.

It provides support for building projects within a JShell session or execute a build script, written in pure Java, in batch mode with:

    jshell build.jsh

The bootstrap overhead from line 3 to 23 (loading `Bach.java` on-the-fly) can be omitted by placing a local version of `Bach.java` into a projects build folder.

For details see https://github.com/sormuras/bach

### CODE REVIEW

**Comment by sormuras** [comment]
It could render the note on regarding [platform adoption](https://github.com/AdoptOpenJDK/jdk9-jigsaw#other-preparations) obsolete...

**Comment by neomatrix369** [comment]
@sormuras thanks for this contribution, this is very interesting stuff.

Can you please make this into a separate exercise under the `session-3-jshell` folder, and add a README.md like the other exercises and put your notes in it. I would like the contents of this repo be as easy to use as possible. 

Have a look at a few exercises/examples to see how they have been laid out.

Thanks.

**Comment by sormuras** [comment]
Sure. I'll move the script and add a README.md. 

On the long run, all `.sh` files will be replaced by `.jsh` scripts. Only need to port some more bash-magic... :-)

**Comment by sormuras** [comment]
I force pushed the new layout for review. If you like it, I'll add build scripts for some more (all?) exercises/examples.

As the bach API is not 100% stable yet, I'll add more commits to branch the coming days. Just want to make sure, the location and layout better now.

File: .gitignore [hunk 1]
```diff
@@ -20,7 +20,7 @@
 *.tar.gz
 *.jimage
 *.jmod
-/target/
+target/
```
Comment (sormuras) [inline .gitignore]: Now, ignores all "target" directories in the tree hierarchy.

**Comment by neomatrix369** [comment]
Thanks please try to add as much documentation and instructions and follow the exercise format so developers and wizz thru the material and have fun learning about this feature.

**Comment by sormuras** [comment]
@neomatrix369 Please review the latest revision I force pushed moments ago.

I simplified the setup to let the user focus on how to build projects with **bach**. I hope the documentation in this PR **exercise-solutions** is detailed enough.

Shall I file another PR against the **master** branch and strip the solutions off the scripts?

**Comment by sormuras** [comment]
Pardon me. The rebase didn't work out like it should. Seems fixed now.

**Comment by sormuras** [comment]
@neomatrix369 What is missing here? Anything I can add/edit/remove?

Shall I file another PR against the **master** branch and replace the solutions with questions and hints?

**Comment by neomatrix369** [comment]
@sormuras Looks good for the `exercise-solutions` version, please do the one for `master` just like you suggested.

I like the fact you have hand-picked only some exercises otherwise it might become a bit overboard to solve all of them.

Thanks again for your time and effort. I'll  merge this one for now and await the `master` version.

**Comment by sormuras** [comment]
@neomatrix369  Thanks!

> As everyone will start using it, there might be requests to iron out the instructions or even PRs with refined versions.

Sure. That's how open source and knowledge works.

> I like the fact you have hand-picked only some exercises otherwise it might become a bit overboard to solve all of them.

I'll add one more example soon, that shows how to resolve external dependencies. An example from the "refactoring-migration" sesson is going to be my target there.

> [...] please do the one for master just like you suggested.

On my way.

Btw. I just added **bach** to the README.md of `session-3-jshell` via #134

**Comment by sormuras** [comment]
Filed https://github.com/AdoptOpenJDK/jdk9-jigsaw/pull/135 against `master`

**Pull request merged**